docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt

```diff
@@ -51,16 +49,176 @@ Resource budget:
+account for the high percentage of trials eliminated after only a short
+time. However, a ``conservative`` adaptive search will only explore a
+small fraction of the configurations explored by an ``aggressive``
+search, given the same budget.
+
+Once the number and types of calls to ASHA are determined (via
+``mode``), the adaptive algorithm will allocate training length budgets
+to the ASHA subroutines, from the overall budget for the adaptive
+algorithm (user-specified through ``budget``). This determines the
+number of trials at each rung (N in the above ASHA example).
+
+Why Asynchronous Halving?
+=========================
+
+Synchronous halving promotes hyperparameter configurations
+synchronously, waiting for each rung to complete before performing any
+promotions. This allows the algorithm to have complete information about
+all trials at the time of promotion, but it results in underutilized
+nodes waiting on completion of validation steps for other
+configurations. ASHA, or asynchronous halving, uses asynchronously
+promotes trials when it has the minimum information required to make a
+decision in order to maximize compute efficiency of the searcher.
 
 See the difference in asynchronous vs. sychronous promotions in the two
 animated GIFs below:
 
 .. image:: /assets/images/sha.gif
 
 .. image:: /assets/images/asha.gif
+
+*****
+ FAQ
+*****
+
+**Q: How do I control how long a trial is trained for before it is
+potentially discarded?**
+
+The training length is guaranteed to be at least ``max_length / 256`` by
+default, or ``max_length / divisor ^ max_rungs-1`` in general. It is
+recommended to configure this in records or epochs if the
+``global_batch_size`` hyperparameter is not constant, to ensure each
+trial trains on the same amount of data.
+
+**Q: How do I make sure ``x`` trials are run the full training length
+(``max_length``)?**
+
+The number of initial trials is determined by a combination of ``mode``,
+``max_trials``, ``divisor``, ``max_rungs``, ``max_length`` and
+``bracket_rungs``. Here is a rule of thumb for the default configuration
+of ``max_rungs: 5`` and ``divisor: 4``, with ``mode: standard`` and a
+large enough ``max_trials``:
+
+-  The initial number of trials is ``max_trials``.
+
+-  To ensure that ``x`` trials are run ``max_length``, set
+   ``max_trials`` high enough for the brackets with their halving rate
+   (the ``divisor``) to allow ``x`` trials to make it to the final
+   ``rungs``. This can be viewed by the command describe below.
+
+A configuration setting that meets set goals can be found by trial and
+error. The command
+
+.. code:: bash
+
+   det preview-search <file_name.yaml>
+
+will display information on the number of trials versus training length
+for the configuration specified in ``file_name.yaml``.
+
+**Q: The adaptive algorithm sounds great so far. What are its
+weaknesses?**
+
+One downside of adaptive is that it results in doing more validations,
```

Comment:
Proposed edit:
In our experience, early-stopping works well across a variety of deep learning models.  However, there may be some search spaces in which early-stopping underperforms simple random search.  This can happen if model complexity varies drastically in a search space leading to different converge rates or if the search space contains hyperparameters that are strongly correlated with training length.